packages/idealo.design/scripts/motifuiImporter.js

```diff
@@ -29,62 +21,134 @@ async function readDirectory(directory) {
+            component.screenshots = screenshots
+        })
+    }
+    return components
 }
 
-if(dangerousUpdateModeArgument){
-    handleImportProcess(pathToMotifUiRepo, motifUiFolder).then(()=>console.log('process finished'))
-}else{
-    console.error('something went wrong')
-}
+async function createFormDataForComponent(components) {
+    for (const component of components) {
+
+        await fs.mkdirSync(localPathToMotifUIScreenshots+'/'+component.screenshotFolderName, (err) => {
+            if (err) {
+                throw err;
+            }
+        });
+        const componentFormData = new FormData();
 
-module.exports.modeArgument = dangerousUpdateModeArgument
+        componentFormData.append('name', component.name);
+        for (const keyword of component.keywords) {
+            componentFormData.append('keywords', keyword);
+        }
 
+        componentFormData.append('screenshotFolderName', component.screenshotFolderName);
+        componentFormData.append('readme', component.readme);
 
+        for (const screenshot of component.screenshots) {
+            const screenshotBuffer = await fs.readFileSync(localScreenshots + '/' + component.screenshotFolderName + '/' + screenshot);
+            const screenshotName= screenshot.replace(/ /g, "_");
+            componentFormData.append('screenshots', screenshotBuffer, screenshotName);
+        }
+        await sendDataToHttpRequest(componentFormData)
+    }
+}
 
+async function sendDataToHttpRequest(componentFormData) {
+    return await axios.put('http://localhost:8080/api/components/update', componentFormData, {
+        headers: componentFormData.getHeaders()
+    });
+}
 
+async function handleImportProcess() {
+    await fse.remove(motifUiFolder)
+        .then(() => console.log('delete motifUI folder successfully'))
+        .catch(err => console.log(err))
+    await fse.copy(pathToMotifUiRepo, motifUiFolder)
+        .then(() => console.log('copy motifUI folder successfully'))
+        .catch(err => console.log(err))
+    await fse.remove(localScreenshots)
+        .then(() => console.log('screenshots folder was removed'))
+        .catch(err => console.log(err))
+    await fse.copy(pathToMotifUIScreenshots, localScreenshots)
+        .then(() => console.log('copy screenshots folder successfully'))
+        .catch(err => console.log(err))
+    await fse.remove(localPathToMotifUIScreenshots)
+        .then(() => console.log('deleted resources folder successfully'))
+        .catch(err => console.log(err))
+    await createPathToMotifUiScreenshots()
 
+    readDirectory(motifUiFolder)
+        .then((subdirectories) => extractComponents(subdirectories))
+        .then((components) => storeScreenshotFolderName(components)
+            .then((components) => storePathToScreenshots(components)
+                .then((components) => createFormDataForComponent(components))))
+}
```

Comment:
This promise chain is hard to read and follow due to the nesting. Could this be refactored into separate variables that you await like above?